### PR TITLE
Elk: new Cask

### PR DESCRIPTION
**Pull request by shuuji3**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

**Comment by shuuji3** [comment]
This is my first Cask submission. If there is any missing item, please let me know. 🙏🏻

File: Casks/elk.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  desc "Native version of Elk, a nimble Mastodon web client"
+  homepage "https://github.com/elk-zone/elk-native"
+
+  livecheck do
+    url "https://github.com/elk-zone/elk-native/releases/"
+    regex(/elk-native-v(\d+(?:\.\d+)+)/i)
+  end
+
+  app "Elk.app"
+end
```
Comment (SMillerDev) [inline Casks/elk.rb:19]: Please add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap) here. The [CreateZap](https://github.com/nrlquaker/homebrew-createzap) tool can help you with this.

File: Casks/elk.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  url "https://github.com/elk-zone/elk-native/releases/download/elk-native-v#{version}/Elk_#{version}_#{arch}.dmg"
+  name "Elk Native"
+  desc "Native version of Elk, a nimble Mastodon web client"
+  homepage "https://github.com/elk-zone/elk-native"
+
+  livecheck do
+    url "https://github.com/elk-zone/elk-native/releases/"
+    regex(/elk-native-v(\d+(?:\.\d+)+)/i)
+  end
```
[the remark this answers is not in the document]
Comment (shuuji3) [reply Casks/elk.rb:16]: Thanks for pointing that out. The reason why I added `livecheck` is that without this, it unintentionally gets the nightly version as the latest like `23.1.16` (https://github.com/elk-zone/elk-native/tags), which causes the version mismatch (expected v0.2.0 vs 23.1.16). So I had to use `elk-native-` to pick up the correct version (`v0.2.0`) here.

I actually read the docs saying that `livecheck` should be avoided if possible. Is there any other way to filter other tags?

File: Casks/elk.rb (hunk 1 above)
Comment (shuuji3) [reply Casks/elk.rb:27]: Create it with CreateZap on https://github.com/shuuji3/homebrew-cask/pull/139778/commits/baca1c2da98c7198bc1aac15224708b6ca3632f7.

**Comment by shuuji3** [comment]
Note: The authors are welcome to add a new cask but they wanted to polish it before merging this. I will upgrade to a new version later and change the PR to ready once they are ready. (ref. https://github.com/elk-zone/elk-native/issues/35)

File: Casks/elk.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (shuuji3) [reply Casks/elk.rb]: FYI: I tried to comment out `livecheck` block to get the actual output:

```shell
> brew audit --cask --online elk
==> Downloading https://github.com/elk-zone/elk-native/releases/download/elk-native-v0.2.1/Elk_0.2.1_macos_aarch64.dmg
==> Downloading from https://objects.githubusercontent.com/github-production-release-asset-2e65be/575943914/70626a25-a514-41a2
######################################################################## 100.0%
audit for elk: failed
 - Version '0.2.1' differs from '23.1.16' retrieved by livecheck.
 - Version '0.2.1' differs from '23.1.16' retrieved by livecheck.
Error: 1 problem in 1 cask detected
```

**Comment by shuuji3** [comment]
OK, I figured out a way to use GithubLatest strategy instead of Git strategy. I needed to override `regex` since elk has the tag with the prefix `elk-native-v` instead of common `v`.

```shell
> brew livecheck elk --debug

Cask:             elk
Livecheckable?:   Yes

URL (url):        https://github.com/elk-zone/elk-native/releases/download/elk-native-v0.2.1/Elk_0.2.1_macos_aarch64.dmg
Strategy:         GithubLatest
Regex:            /href=.*?\/tag\/elk-native-v?(\d+(?:\.\d+)+)["' >]/i
URL (strategy):   https://github.com/elk-zone/elk-native/releases/latest
URL (final):      https://github.com/elk-zone/elk-native/releases/tag/elk-native-v0.2.1

Matched Versions:
0.2.1

elk: 0.2.1 ==> 0.2.1
```

```shell
> brew audit --cask --online elk
==> Downloading https://github.com/elk-zone/elk-native/releases/download/elk-native-v0.2.1/Elk_0.2.1_macos_aarch64.dmg
Already downloaded: /Users/shuuji3/Library/Caches/Homebrew/downloads/049bcccff5858587dc26f1b62415cb56541ddf5596d77d24aad9851f850e5817--Elk_0.2.1_macos_aarch64.dmg
audit for elk: passed
```

```shell
> brew style --fix elk

1 file inspected, no offenses detected
```

**Comment by miccal** [comment]
Thank you @shuuji3.

**Review by miccal — APPROVED** [review]

**Pull request merged**